commit c2a8653c197d67f8ad563f5417f2e9bcaad913f3
Author: Luca Weiss <luca.weiss@fairphone.com>
Date:   Fri Dec 1 10:33:18 2023 +0100

    media: venus: core: Set up secure memory ranges for SC7280
    
    Not all SC7280 devices ship with ChromeOS firmware. Other devices need
    PAS for image authentication. That requires the predefined virtual
    address ranges to be passed via scm calls. Define them to enable Venus
    on non-CrOS SC7280 devices.
    
    Reviewed-by: Konrad Dybcio <konrad.dybcio@linaro.org>
    Reviewed-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
    Reviewed-by: Vikash Garodia <quic_vgarodia@quicinc.com>
    Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>
    Signed-off-by: Stanimir Varbanov <stanimir.k.varbanov@gmail.com>
    Signed-off-by: Hans Verkuil <hverkuil-cisco@xs4all.nl>

diff --git a/drivers/media/platform/qcom/venus/core.c b/drivers/media/platform/qcom/venus/core.c
index 9cffe975581b..a712dd4f02a5 100644
--- a/drivers/media/platform/qcom/venus/core.c
+++ b/drivers/media/platform/qcom/venus/core.c
@@ -881,6 +881,10 @@ static const struct venus_resources sc7280_res = {
 	.vmem_size = 0,
 	.vmem_addr = 0,
 	.dma_mask = 0xe0000000 - 1,
+	.cp_start = 0,
+	.cp_size = 0x25800000,
+	.cp_nonpixel_start = 0x1000000,
+	.cp_nonpixel_size = 0x24800000,
 	.fwname = "qcom/vpu-2.0/venus.mbn",
 };